### PR TITLE
Prioritize the cuda builds if the user can using build numbers and add cudart dependency

### PR DESCRIPTION
**Pull request by hmaarrfk**
I am proposing two things:

1. Bumping the build number of the cuda builds so that the solver prioritizes them. On my machines, the solver was choosing the CPU builds by default even though I had the `__cuda` virtual package.

2. ~Harmonizing the build string with that of tensorflow and pytorch. This is likely a larger breaking change but I think it will help translate people's learnings from one package to an other.~

Closes: https://github.com/conda-forge/onnxruntime-feedstock/issues/93
Thanks!

cc: @jefromson

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by traversaro** [comment]
> 2\. Harmonizing the build string with that of tensorflow and pytorch. This is likely a larger breaking change but I think it will help translate people's learnings from one package to an other.

I do not strongly prefer one convention or another, but just to give a bit of context: I was not sure on the build string style to use back on when I enable CUDA in this package, see https://github.com/conda-forge/onnxruntime-feedstock/pull/63#issuecomment-1614740936 and the linked comment https://github.com/conda-forge/librealsense-feedstock/issues/19#issuecomment-1445464186 . Looking at that comment, it seems that I was inspired by the build strings used by `arrow-cpp` and `jaxlib`.

**Comment by hmaarrfk** [comment]
> I do not strongly prefer one convention or another, 

Honestly, after thinking about it for a while, I somewhat do not want to even try to "standardize" this since it would lead to a false sense of "consistency". I don't think that people should be using:

```
onnxruntime=*=*cuda
```
or 

```
onnxruntime=*=cuda*
```
and would prefer that mamba just "finds it better".

I'll revert the build string changes.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -61,10 +70,13 @@ requirements:
...
-    - libcublas-dev     # [(cuda_compiler_version or "").startswith("12")]
-    - libcusparse-dev   # [(cuda_compiler_version or "").startswith("12")]
-    - libcurand-dev     # [(cuda_compiler_version or "").startswith("12")]
-    - libcufft-dev      # [(cuda_compiler_version or "").startswith("12")]
+    {% if cuda_major >= 12 %}
+    - libcublas-dev
+    - libcusparse-dev
+    - libcurand-dev
+    - libcufft-dev
+    - cuda-cudart-dev
```
Comment (hmaarrfk) [inline recipe/meta.yaml:78]: I think this was missing and causing the issues in 

https://github.com/conda-forge/onnxruntime-feedstock/issues/93

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Review by xhochy — APPROVED** [review]

**Pull request merged**